message_app_client/src/state.js

```diff
@@ -3,13 +3,16 @@ import io from 'socket.io-client';
 // When needed, we will pass a custom api URL (test env, prod env) using .env
 const apiURL = process.env.API_URL || 'http://127.0.0.1:5000' ;
 
-export const socket = io(`${apiURL}messages`, { transports: ["websocket"] });
+export const socket = io(`${apiURL}/messages`, { transports: ["websocket"] });
 
-export const getApiRoute = (name) => {
-    return `${apiURL}/${apiRoutes[name]}`;
+export const getApiRoute = (name, parameter=null) => {
+    let URL = `${apiURL}/${apiRoutes[name]}`;
+    if (parameter !== null) URL = URL + `/${parameter}`;
+    return URL;
```

Comment:
Which param should we expect here if it is ever used? This function is bad for two reasons:
- It does not specify what format the parameter would be
- It assumes that there will be only one parameter
- The parameter for each endpoint is very specific that to endpoint, and here we are doing it in a generic function to get the API route we want.

There is a mechanism to specify what param(s) an endpoint will expect, so that only expected params will be processed. What an endpoint expects should be specified at that endpoint, not here.

We can implement this logic ourselves or use a library. For now, please keep the original version of the `getApiRoute` function.

Reply:
I will change it back to the original version in the next commit.